gemrb/plugins/AREImporter/AREImporter.cpp

```diff
@@ -478,7 +478,15 @@ Map* AREImporter::GetMap(const char *ResRef, bool day_or_night)
 	for (i = 0; i < MAX_RESCOUNT; i++) {
 		str->ReadDword( map->SongHeader.SongList + i );
 	}
-	str->Seek( RestHeader + 32, GEM_STREAM_START );
+
+	if (core->HasFeature(GF_PST_STATE_FLAGS)) {
+		str->Seek( RestHeader + 32, GEM_STREAM_START );
+		str->ReadDword(&map->SongHeader.reverbID);
+	} else {
+		map->SongHeader.reverbID = EFX_PROFILE_REVERB_INVALID;
+	}
```

Comment:
The seeks look odd, am I missing something?
1. The non-pst case also needs to seek forward for the next reads to start at the right location.
2. The rest header is a separate structure, so I don't think it's reading the reverb id either. It looks 48 bytes ahead of the last song resref to me:
https://gibberlings3.github.io/iesdp/file_formats/ie_formats/are_v1.htm#formAREAV1_0_Song_entries

So I suggest you put the original seek after the whole if block and add a new one to get the reverb id to the pst case. Or use the if only to reset the id for non-pst games, after reading the value.

Reply:
Oh, mistake, yes.

Reply:
SongHeader + 0x50 should be correct, rest-parsing was misused.